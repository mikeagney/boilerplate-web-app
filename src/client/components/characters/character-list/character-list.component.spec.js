import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CharacterList from './character-list.component';

describe('CharacterList component', () => {
  const defaultProps = {
    characterIds: ['a1a', 'b2b'],
    selectedId: 'b2b',
  };

  it('will render Character components with the appropriate character selected', () => {
    // Arrange
    const props = defaultProps;

    // Act
    const wrapper = shallow(<CharacterList {...props} />);

    // Assert
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
