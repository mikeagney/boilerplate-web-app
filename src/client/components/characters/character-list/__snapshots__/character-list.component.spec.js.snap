// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component will render Character components with the appropriate character selected 1`] = `
<ForwardRef
  activeKey="b2b"
  className="character-list"
  onSelect={[MockFunction]}
>
  <Bootstrap(Row)>
    <Col
      as="div"
      sm={3}
    >
      <Nav
        as="div"
        className="flex-column"
        fill={false}
        justify={false}
        variant="pills"
      >
        <NavLink
          as={[Function]}
          disabled={false}
          eventKey="a1a"
          key="a1a"
        >
          Mock Character 1
        </NavLink>
        <NavLink
          as={[Function]}
          disabled={false}
          eventKey="b2b"
          key="b2b"
        >
          Mock Character 2
        </NavLink>
        <NavLink
          as={[Function]}
          disabled={false}
          eventKey=""
        >
          <MdAdd />
        </NavLink>
      </Nav>
    </Col>
    <Col
      as="div"
      sm={9}
    >
      <Bootstrap(TabContent)>
        <TabPane
          eventKey="a1a"
          key="a1a"
        >
          <Connect(Character)
            characterId="a1a"
            selected={false}
          />
        </TabPane>
        <TabPane
          eventKey="b2b"
          key="b2b"
        >
          <Connect(Character)
            characterId="b2b"
            selected={true}
          />
        </TabPane>
        <TabPane
          eventKey=""
        >
          <Connect(NewCharacter) />
        </TabPane>
      </Bootstrap(TabContent)>
    </Col>
  </Bootstrap(Row)>
</ForwardRef>
`;
